### PR TITLE
Assert that sent funds match assets

### LINKED ISSUE #9: Amount of native coins sent to `BalancingProvideLiquidity` is never validated
We only do `assert_only_native_coins` but never check that the amount of native tokens sent is the same as in the AssetList. This is low severity since this contract should not be holding any funds, but if there were to be funds donated to the contract someone could steal them by passing them in the AssetList but not in the funds parameter.

### PR DESCRIPTION
**Pull request by apollo-pacman**
closes #9

### CODE REVIEW

**Comment by apollo-pacman** [comment]
@apollo-sturdy why do we use the assets AssetList in the osmosis case? Could we not just do `let assets: AssetList = info.funds.into()` instead? and leave the assets argument unused.

File: contracts/osmosis-liquidity-helper/src/contract.rs [hunk 1]
```diff
@@ -88,6 +88,14 @@ pub fn execute_balancing_provide_liquidity(
     // Assert that only native coins are sent
     assert_only_native_coins(&assets)?;
```
Comment (apollo-sturdy) [inline contracts/osmosis-liquidity-helper/src/contract.rs:89]: Could remove this now since the check below would fail on the same conditions.

Comment (apollo-pacman) [reply contracts/osmosis-liquidity-helper/src/contract.rs:89]: Ill go for [this](https://github.com/apollodao/liquidity-helpers/pull/13#issuecomment-1351331886) solution instead

**Comment by apollo-sturdy** [comment]
> @apollo-sturdy why do we use the assets AssetList in the osmosis case? Could we not just do `let assets: AssetList = info.funds.into()` instead? and leave the assets argument unused.

Yeah I don't see why not. The only downside there is that it won't fail explicitly if someone tries to send more assets in the assetlist than they send in funds, which is maybe fine, just not as explicit that it didn't do what you thought it would.

**Comment by apollo-pacman** [comment]
> > @apollo-sturdy why do we use the assets AssetList in the osmosis case? Could we not just do `let assets: AssetList = info.funds.into()` instead? and leave the assets argument unused.
> 
> Yeah I don't see why not. The only downside there is that it won't fail explicitly if someone tries to send more assets in the assetlist than they send in funds, which is maybe fine, just not as explicit that it didn't do what you thought it would.

good point! Let's keep this solution to be explicit with the error. I'll address your comment above and merge this

File: contracts/osmosis-liquidity-helper/src/contract.rs (hunk 1 above)
Comment (apollo-pacman) [reply contracts/osmosis-liquidity-helper/src/contract.rs]: fixed

**Pull request merged**